tests/nativeapp/test_version_create_processor.py

```diff
@@ -682,3 +684,94 @@ def test_process_existing_release_directives_w_existing_version_two(
     mock_sync.assert_called_once()
     assert mock_existing_version_info.call_count == 2
     mock_add_patch.assert_called_once()
+
+
+# Test version create when the app package doesn't have the magic CLI comment
+@mock.patch(FIND_VERSION_FROM_MANIFEST, return_value=("manifest_version", None))
+@mock.patch(f"{VERSION_MODULE}.check_index_changes_in_git_repo", return_value=None)
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}.create_app_package",
+    side_effect=ApplicationPackageAlreadyExistsError(""),
+)
+@mock.patch(NATIVEAPP_MANAGER_EXECUTE)
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}._apply_package_scripts", return_value=None
+)
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}.sync_deploy_root_with_stage",
+    return_value=None,
+)
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}.get_existing_release_directive_info_for_version",
+    return_value=None,
+)
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}.get_existing_version_info", return_value=None
+)
+@mock.patch(f"{VERSION_MODULE}.{CREATE_PROCESSOR}.add_new_version", return_value=None)
+@mock.patch(f"snowflake.cli.plugins.nativeapp.policy.{TYPER_CONFIRM}")
```

Comment:
Possibly controversial opinion here: this is way too many mocks. I think the way we write tests makes them brittle when we refactor the code, and is just too much work overall. Possible solutions I can think of: 1) leave this kind of test to integ tests, 2) invest in setting up a good fixture that is used by most tests so that all the patching is done in one place (with reasonable defaults) so that each test only has to override the behaviour it needs. Thoughts? Could be a good PL topic too.

Reply:
yeah, i agree there's a lot of copypasta of mocks going on. I think there's still some value in unit tests (especially since they're faster and easier to run locally), so I think your second suggestion is easily doable with `autouse` fixtures at the file or package level